Build fails on Gentoo Linux (fix included)

**Describe the bug**
Build using `build-debug` fails on Gentoo Linux (amd64, stable, desktop profile) due to errors related to zlib:

```
In file included from ../include/zip.h:54,
                 from dosbox.cpp:77:
../include/ioapi.h:132:51: error: expected initializer before ‘OF’
  132 | typedef voidpf   (ZCALLBACK *open_file_func)      OF((voidpf opaque, const char* filename, int mode));
      |                                                   ^~
../include/ioapi.h:133:51: error: expected initializer before ‘OF’
  133 | typedef uLong    (ZCALLBACK *read_file_func)      OF((voidpf opaque, voidpf stream, void* buf, uLong size));
      |                                                   ^~
../include/ioapi.h:134:51: error: expected initializer before ‘OF’
  134 | typedef uLong    (ZCALLBACK *write_file_func)     OF((voidpf opaque, voidpf stream, const void* buf, uLong size));
      |                                                   ^~
../include/ioapi.h:135:51: error: expected initializer before ‘OF’
  135 | typedef int      (ZCALLBACK *close_file_func)     OF((voidpf opaque, voidpf stream));
      |                                                   ^~
../include/ioapi.h:136:51: error: expected initializer before ‘OF’
  136 | typedef int      (ZCALLBACK *testerror_file_func) OF((voidpf opaque, voidpf stream));
      |                                                   ^~
../include/ioapi.h:138:51: error: expected initializer before ‘OF’
  138 | typedef long     (ZCALLBACK *tell_file_func)      OF((voidpf opaque, voidpf stream));
      |                                                   ^~
../include/ioapi.h:139:51: error: expected initializer before ‘OF’
  139 | typedef long     (ZCALLBACK *seek_file_func)      OF((voidpf opaque, voidpf stream, uLong offset, int origin));
      |                                                   ^~
../include/ioapi.h:145:5: error: ‘open_file_func’ does not name a type
  145 |     open_file_func      zopen_file;
      |     ^~~~~~~~~~~~~~
../include/ioapi.h:146:5: error: ‘read_file_func’ does not name a type
  146 |     read_file_func      zread_file;
      |     ^~~~~~~~~~~~~~
../include/ioapi.h:147:5: error: ‘write_file_func’ does not name a type
  147 |     write_file_func     zwrite_file;
      |     ^~~~~~~~~~~~~~~
../include/ioapi.h:148:5: error: ‘tell_file_func’ does not name a type
  148 |     tell_file_func      ztell_file;
      |     ^~~~~~~~~~~~~~
../include/ioapi.h:149:5: error: ‘seek_file_func’ does not name a type
  149 |     seek_file_func      zseek_file;
      |     ^~~~~~~~~~~~~~
../include/ioapi.h:150:5: error: ‘close_file_func’ does not name a type
  150 |     close_file_func     zclose_file;
      |     ^~~~~~~~~~~~~~~
../include/ioapi.h:151:5: error: ‘testerror_file_func’ does not name a type
  151 |     testerror_file_func zerror_file;
      |     ^~~~~~~~~~~~~~~~~~~
../include/ioapi.h:155:51: error: expected initializer before ‘OF’
  155 | typedef ZPOS64_T (ZCALLBACK *tell64_file_func)    OF((voidpf opaque, voidpf stream));
      |                                                   ^~

[and many more]
```

There's a rather simple fix:
```
diff -Naurp a/include/ioapi.h b/include/ioapi.h
--- a/include/ioapi.h	2020-08-03 21:18:18.606933344 +0200
+++ b/include/ioapi.h	2020-08-03 21:19:30.329932231 +0200
@@ -105,6 +105,10 @@ typedef unsigned long long int ZPOS64_T;
 extern "C" {
 #endif
 
+#ifdef _Z_OF
+#undef OF
+#define OF _Z_OF
+#endif
 
 #define ZLIB_FILEFUNC_SEEK_CUR (1)
 #define ZLIB_FILEFUNC_SEEK_END (2)
```

With this fix applied, DOSBox-X builds fine without any errors.

Source of fix: [issue #26 of project godotengine/godot here on GitHub](https://github.com/godotengine/godot/issues/26#issuecomment-35689499). It would seem this issue stems from [Gentoo sanitizing zlib's headers](https://bugs.gentoo.org/383179). I'm not sure if this fix is safe for other build environments (comments of the issue linked above suggests it should be).


**To Reproduce**
On Gentoo Linux, run the following commands:
```
# git clone https://github.com/joncampbell123/dosbox-x.git dosbox-x
# cd dosbox-x
# ./build-debug
```

**Expected behavior**
DOSBox-X should build without errors.

**Screenshots**
(none)

**Environment (please complete the following information):**
 - OS: Gentoo Linux (amd64, stable, desktop profile)
 - zlib package: sys-libs/zlib-1.2.11-r2
 - DOSBox-X sources: GIT commit c52a3aa082af2536b25d9990622e8520c11d19a0

**Additional context**
(none)

--- a/src/dos/cdrom.h
+++ b/src/dos/cdrom.h
@@ -438,26 +438,26 @@ class CDROM_Interface_Aspi : public CDROM_Interface
 public:
 	virtual ~CDROM_Interface_Aspi(void);
 
-	bool	SetDevice			(char* path, int forceCD);
+	bool	SetDevice			(char* path, int forceCD) override;
 
-	bool	GetUPC				(unsigned char& attr, char* upc);
+	bool	GetUPC				(unsigned char& attr, char* upc) override;
 
-	bool	GetAudioTracks		(int& stTrack, int& end, TMSF& leadOut);
-	bool	GetAudioTrackInfo	(int track, TMSF& start, unsigned char& attr);
-	bool	GetAudioSub			(unsigned char& attr, unsigned char& track, unsigned char& index, TMSF& relPos, TMSF& absPos);
-	bool	GetAudioStatus		(bool& playing, bool& pause);
-	bool	GetMediaTrayStatus	(bool& mediaPresent, bool& mediaChanged, bool& trayOpen);
+	bool	GetAudioTracks		(int& stTrack, int& end, TMSF& leadOut) override;
+	bool	GetAudioTrackInfo	(int track, TMSF& start, unsigned char& attr) override;
+	bool	GetAudioSub			(unsigned char& attr, unsigned char& track, unsigned char& index, TMSF& relPos, TMSF& absPos) override;
+	bool	GetAudioStatus		(bool& playing, bool& pause) override;
+	bool	GetMediaTrayStatus	(bool& mediaPresent, bool& mediaChanged, bool& trayOpen) override;
 
-	bool	PlayAudioSector		(unsigned long start,unsigned long len);
-	bool	PauseAudio			(bool resume);
-	bool	StopAudio			(void);
-	void	ChannelControl		(TCtrl ctrl) { (void)ctrl; return; };
+	bool	PlayAudioSector		(unsigned long start,unsigned long len) override;
+	bool	PauseAudio			(bool resume) override;
+	bool	StopAudio			(void) override;
+	void	ChannelControl		(TCtrl ctrl) override { (void)ctrl; return; };
 	
-	bool	ReadSectors			(PhysPt buffer, bool raw, unsigned long sector, unsigned long num);
+	bool	ReadSectors			(PhysPt buffer, bool raw, unsigned long sector, unsigned long num) override;
 	/* This is needed for IDE hack, who's buffer does not exist in DOS physical memory */
-	bool	ReadSectorsHost			(void* buffer, bool raw, unsigned long sector, unsigned long num);
+	bool	ReadSectorsHost			(void* buffer, bool raw, unsigned long sector, unsigned long num) override;
 	
-	bool	LoadUnloadMedia		(bool unload);
+	bool	LoadUnloadMedia		(bool unload) override;
 	
 private:
 	DWORD	GetTOC				(LPTOC toc);
@@ -491,29 +491,29 @@ public:
 	CDROM_Interface_Ioctl		(cdioctl_cdatype ioctl_cda);
 	virtual ~CDROM_Interface_Ioctl(void);
 
-	bool	SetDevice			(char* path, int forceCD);
+	bool	SetDevice			(char* path, int forceCD) override;
 
-	bool	GetUPC				(unsigned char& attr, char* upc);
+	bool	GetUPC				(unsigned char& attr, char* upc) override;
 
-	bool	GetAudioTracks		(int& stTrack, int& end, TMSF& leadOut);
-	bool	GetAudioTrackInfo	(int track, TMSF& start, unsigned char& attr);
-	bool	GetAudioSub			(unsigned char& attr, unsigned char& track, unsigned char& index, TMSF& relPos, TMSF& absPos);
-	bool	GetAudioStatus		(bool& playing, bool& pause);
-	bool	GetMediaTrayStatus	(bool& mediaPresent, bool& mediaChanged, bool& trayOpen);
+	bool	GetAudioTracks		(int& stTrack, int& end, TMSF& leadOut) override;
+	bool	GetAudioTrackInfo	(int track, TMSF& start, unsigned char& attr) override;
+	bool	GetAudioSub			(unsigned char& attr, unsigned char& track, unsigned char& index, TMSF& relPos, TMSF& absPos) override;
+	bool	GetAudioStatus		(bool& playing, bool& pause) override;
+	bool	GetMediaTrayStatus	(bool& mediaPresent, bool& mediaChanged, bool& trayOpen) override;
 
-	bool	PlayAudioSector		(unsigned long start,unsigned long len);
-	bool	PauseAudio			(bool resume);
-	bool	StopAudio			(void);
-	void	ChannelControl		(TCtrl ctrl);
+	bool	PlayAudioSector		(unsigned long start,unsigned long len) override;
+	bool	PauseAudio			(bool resume) override;
+	bool	StopAudio			(void) override;
+	void	ChannelControl		(TCtrl ctrl) override;
 	
 	bool	ReadSector			(uint8_t *buffer, bool raw, unsigned long sector);
-	bool	ReadSectors			(PhysPt buffer, bool raw, unsigned long sector, unsigned long num);
+	bool	ReadSectors			(PhysPt buffer, bool raw, unsigned long sector, unsigned long num) override;
 	/* This is needed for IDE hack, who's buffer does not exist in DOS physical memory */
-	bool	ReadSectorsHost			(void* buffer, bool raw, unsigned long sector, unsigned long num);
+	bool	ReadSectorsHost			(void* buffer, bool raw, unsigned long sector, unsigned long num) override;
 	
-	bool	LoadUnloadMedia		(bool unload);
+	bool	LoadUnloadMedia		(bool unload) override;
 
-	void	InitNewMedia		(void) { Close(); Open(); };
+	void	InitNewMedia		(void) override { Close(); Open(); };
 private:
 
 	bool	Open				(void);

--- a/src/dos/dos_programs.cpp
+++ b/src/dos/dos_programs.cpp
@@ -3221,7 +3221,7 @@ static void LOADROM_ProgramStart(Program * * make) {
 #if C_DEBUG
 class BIOSTEST : public Program {
 public:
-    void Run(void) {
+    void Run(void) override {
         if (!(cmd->FindCommand(1, temp_line))) {
             WriteOut("Must specify BIOS file to load.\n");
             return;
@@ -7196,7 +7196,7 @@ void VFRCRATE_ProgramStart(Program * * make);
 #if defined C_DEBUG
 class NMITEST : public Program {
 public:
-    void Run(void) {
+    void Run(void) override {
         if (cmd->FindExist("-?", false) || cmd->FindExist("/?", false)) {
 			WriteOut("Generates a non-maskable interrupt (NMI).\n\nNMITEST\n\nNote: This is a debugging tool to test if the interrupt handler works properly.\n");
             return;

--- a/src/hardware/ipx.cpp
+++ b/src/hardware/ipx.cpp
@@ -1003,7 +1003,7 @@ public:
 		}
 	}
 
-	void Run(void)
+	void Run(void) override
 	{
 		WriteOut("IPX Tunneling utility for DOSBox-X\n\n");
 		if(!cmd->GetCount()) {

